google_guest_agent/network/manager/dhclient_linux.go

```diff
@@ -0,0 +1,191 @@
+	primaryIface := payload[0]
+	primaryNic, err := GetInterfaceByMAC(primaryIface.Mac)
+	if err != nil {
+		return fmt.Errorf("error getting primary network interface: %s", err)
+	}
+	if primaryIface.DHCPv6Refresh != "" {
+		// Wait for tentative IPs as part of SLAAC to resolve for primary interface.
+		tentative := []string{"-6", "-o", "a", "s", "dev", primaryNic.Name, "scope", "link", "tentative"}
+		for i := 0; i < 5; i++ {
+			res := run.WithOutput(ctx, "ip", tentative...)
+			if res.ExitCode == 0 && res.StdOut == "" {
+				break
+			}
+			time.Sleep(1 * time.Second)
+		}
+	}
+
+	for _, iface := range googleIpv6Interfaces {
+		// Set the appropriate system values.
+		val := fmt.Sprintf("net.ipv6.conf.%s.accept_ra_rt_info_max_plen=128", iface)
+		if err := run.Quiet(ctx, "sysctl", val); err != nil {
+			return err
+		}
+	}
+	switch {
+	case primaryIface.DHCPv6Refresh == "":
+		// disable -- release the IPv6 lease.
+		if err := run.Quiet(ctx, "dhclient", "-r", "-6", "-1", "-v", primaryNic.Name); err != nil {
+			logger.Errorf("error configuring IPv6: %v", err)
+		}
+	case primaryIface.DHCPv6Refresh != "":
+		// enable
+		if err := run.Quiet(ctx, "dhclient", "-1", "-6", "-v", primaryNic.Name); err != nil {
+			logger.Errorf("error configuring IPv6: %v", err)
+		}
+
+	}
+
+	// Set up IPV4 for all secondary interfaces.
+	for _, iface := range googleInterfaces[1:] {
+		if err := runDhclient(ctx, ipv4, iface); err != nil {
+			logger.Errorf("failed to run dhclient: %+x", err)
+		}
+	}
+
+	// Setup IPV6 for all interfaces (primary included).
+	for _, iface := range googleIpv6Interfaces {
+		if err := runDhclient(ctx, ipv6, iface); err != nil {
+			logger.Errorf("failed to run dhclient: %+x", err)
+		}
+	}
+	return nil
+}
+
+// runDhclient releases and re-obtains a lease with the provided IP version
+// for the given network interface.
+func runDhclient(ctx context.Context, ipVersion ipVersion, nic string) error {
+	pidFile := path.Join(pidFileDir, fmt.Sprintf("dhclient.google-guest-agent.%s.pid", nic))
+	dhclientArgs := []string{ipVersion.dhclientArg, "-pf", pidFile, nic}
+
+	releaseArgs := append(dhclientArgs, "-r", nic)
+
+	// Always release the lease before trying to run dhclient again.
+	if err := run.Quiet(ctx, "dhclient", releaseArgs...); err != nil {
+		return fmt.Errorf("error releasing lease for %s: %v", nic, err)
+	}
+
+	// Now obtain a lease.
+	if err := run.Quiet(ctx, "dhclient", dhclientArgs...); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Rollback releases all leases from DHClient, effectively undoing the dhclient configurations.
+func (n dhclient) Rollback(ctx context.Context, payload []metadata.NetworkInterfaces) error {
+	// Find the dhclient processes.
+	ifaces, err := interfaceNames(payload)
+	if err != nil {
+		return fmt.Errorf("failed to get interfaces: %s", err)
+	}
+
+	// Release all the interface leases from dhclient.
+	for _, iface := range ifaces {
+		if err = run.Quiet(ctx, "dhclient", "-r", iface); err != nil {
```

Comment:
same as above: we need to specify pid file here

Reply:
Done.